### PR TITLE
feat(cli): Add cli flag for (optionally) selecting the config file

### PR DESCRIPTION
**Pull request by Teufelchen1**
Hi! 👋 

This is an example on how one could add an optional configuration file flag to the cli.

I'm not sure if I nailed the usage of clap - specifying the flag twice (per `check` and `run`) feels odd.

This change should not break existing setups - since I do not have one, I could not test that claim.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -25,6 +25,15 @@ pub enum SubCommand {
         /// Write valid config as website-stalker.yaml
         #[arg(long)]
         rewrite_yaml: bool,
+
+        /// Configuration file to use, default website-stalker.yaml
```
Comment (EdJoPaTo) [inline src/cli.rs:29]: The default will already be shown by clap, no need to put it in the docs too
```suggestion
        /// Configuration file to use
```

**Review by EdJoPaTo — COMMENTED** [review]
I'm not sure yet whether I like having a configurable config file path or not. Having it basically means the `sites` folder path should be configurable too. Currently the folder setup with the `website-stalker.yaml` and the `sites` folder is strict and therefore relatively easy to handle with the git integration as there are not so many sideeffects possible. But its also nice to have more things configurable.

---

> specifying the flag twice (per check and run) feels odd.

this could be done as a global flag and used in all the subcommands.

File: src/cli.rs [hunk 2]
```diff
@@ -25,6 +25,15 @@ pub enum SubCommand {
...
+        #[arg(
+            short = 'c',
+            long = "config",
+            value_hint = ValueHint::FilePath,
+            default_value = "website-stalker.yaml",
+        )]
+        config_name: String,
```
[the remark this answers is not in the document]
Comment (Teufelchen1) [reply src/cli.rs]: Oh nice! Thanks for the hint! I adapted it accordingly.

**Comment by Teufelchen1** [comment]
>> specifying the flag twice (per check and run) feels odd.
>
>this could be done as a global flag and used in all the subcommands.

Neat! I adopted that change.


> I'm not sure yet whether I like having a configurable config file path or not. Having it basically means the sites folder path should be configurable too. Currently the folder setup with the website-stalker.yaml and the sites folder is strict and therefore relatively easy to handle with the git integration as there are not so many sideeffects possible. But it's also nice to have more things configurable.

Hm, I see. I might have rushed this and spend too little time thinking of the side-effects of this change. 
For the moment, closing this PR seems the best action for me. We can always come back later, when we are certain this change is useful and the side effects are acceptable / accounted for.

**Comment by EdJoPaTo** [comment]
> Hm, I see. I might have rushed this and spend too little time thinking of the side-effects of this change. For the moment, closing this PR seems the best action for me. We can always come back later, when we are certain this change is useful and the side effects are acceptable / accounted for.

Guess that's also a benefit of having multiple people thinking about things. There always will be different goals and benefits so I don't think it's a bad thing to suggest things and to start thinking about a topic.

It's the same in this case: I do like the idea of more customisability in general but after thinking about it it does not seem that wise in this case. Thinking about it made it more clear that that's the case for this tool.

Personally I try to start a discussion first before providing a PQ so I get other mindsets involved. Often not via GitHub (depending on the project). Then I try to work on the improvements with higher confidence that they will be merged. That works kinda well for me currently. Maybe that approach also seems helpful to you.

As you suggested I will close this. Feel free to reopen / share thoughts when it feels relevant so we can talk about it.

Thanks again for your thoughts and efforts to improve this tool!

**Pull request closed without merging**